src/screens/staking-delegate-form.tsx

```diff
@@ -19,39 +20,42 @@ export const StakingDelegateFormScreen = () => {
   const wallet = Wallet.getById(account);
   const balances = useWalletsBalance([wallet!]);
   const currentBalance = useMemo(() => balances[account], [balances, account]);
-  const [fee, setFee] = useState<Balance | null>(null);
+  const [fee, _setFee] = useState<Balance | null>(null);
 
   const setDefaultFee = useCallback(
-    () => setFee(new Balance(Cosmos.fee.amount)),
+    () => _setFee(new Balance(Cosmos.fee.amount)),
     [],
   );
 
-  useLayoutEffectAsync(async () => {
-    const timer = setTimeout(() => {
-      setDefaultFee();
-    }, FEE_ESTIMATING_TIMEOUT_MS);
-
-    const instance = await getProviderInstanceForWallet(wallet!, true);
+  const setFee = useThrottle(async (amount?: string) => {
     const cosmos = new Cosmos(app.provider);
+    const instance = await getProviderInstanceForWallet(wallet!, true);
+
     try {
-      setFee(null);
+      _setFee(null);
       const f = await cosmos.simulateDelegate(
         instance,
         wallet!.path!,
         validator.operator_address,
-        currentBalance.availableForStake,
+        amount ? new Balance(amount) : currentBalance.availableForStake,
       );
       Logger.log('f.amount', f.amount);
-      setFee(new Balance(f.amount));
+      _setFee(new Balance(f.amount));
     } catch (err) {
       if (instance instanceof ProviderLedgerReactNative) {
         instance.abort();
-        clearTimeout(timer);
         setDefaultFee();
       }
-    } finally {
-      clearTimeout(timer);
     }
+  }, 500);
+
+  useLayoutEffectAsync(async () => {
+    const timer = setTimeout(() => {
+      setDefaultFee();
+    }, FEE_ESTIMATING_TIMEOUT_MS);
+
+    await setFee();
```

Comment:
I think we need to have `Estimating...` state in our UI, so it's better to use useEffect here.

Reply:
This logic incapsulate inside NetworkFee component. Each time will fee calculation happens Network fee shows Estimating... and button disabled too.